File: brain/e3cca4c1-456c-4150-baee-d6bd459f953a/scratch/check_tags.py

import sys

def check_tags(content):
    tags = ['div', 'main', 'section', 'button', 'span', 'h1', 'h2', 'h3', 'h4', 'StepWrapper', 'motion.div', 'AnimatePresence', 'motion.button']
    counts = {t: 0 for t in tags}
    
    # Very crude counting
    for t in tags:
        counts[t] = content.count(f'<{t}') - content.count(f'</{t}')
    
    return counts

with open(r'd:\ezoflife\frontend\src\modules\user\pages\HomePage.jsx', 'r', encoding='utf-8') as f:
    content = f.read()
    print(check_tags(content))

File: brain/e3cca4c1-456c-4150-baee-d6bd459f953a/scratch/check_tags.js

const fs = require('fs');
const content = fs.readFileSync('d:\\ezoflife\\frontend\\src\\modules\\user\\pages\\HomePage.jsx', 'utf8');

const tags = ['div', 'main', 'section', 'button', 'span', 'h1', 'h2', 'h3', 'h4', 'StepWrapper', 'motion.div', 'AnimatePresence', 'motion.button'];
const counts = {};

tags.forEach(t => {
    const opening = (content.match(new RegExp('<' + t.replace('.', '\\.'), 'g')) || []).length;
    const closing = (content.match(new RegExp('</' + t.replace('.', '\\.'), 'g')) || []).length;
    const selfClosing = (content.match(new RegExp('<' + t.replace('.', '\\.') + '[^>]*/>', 'g')) || []).length;
    counts[t] = opening - closing - selfClosing;
});

console.log(counts);
